odoo/addons/base/res/res_config.py

```diff
@@ -536,6 +585,25 @@ def execute(self):
                 groups.write({'implied_ids': [(3, implied_group.id)]})
                 implied_group.write({'users': [(3, user.id) for user in groups.mapped('users')]})
 
+        # config fields: store ir.config_parameters
+        SuIrConfigParameter = self.env['ir.config_parameter'].sudo()
+        for name, icp in classified['config']:
+            value = self[name]
+            if value and self._fields[name].type == 'char':
+                # storing developer keys as ir.config_parameter may lead to nasty
+                # bugs when users leave spaces around them
+                value = value.strip()
+            elif value and self._fields[name].type in ['integer', 'float']:
+                value = repr(value)
+            elif value and self._fields[name].type == 'many2one':
+                value = value.id
+            elif not value and self._fields[name].type == 'many2one':
+                # /!\ tricky:
+                # - value can be falsy in case of a many2one but being an empty recordset
+                # - we actually want to send the boolean False to set_param
+                value = False
```

Comment:
Overcomplicated, IMHO. If the recordset is empty, its `id` is `False`.
So when the field type is `many2one`, this always works:
```python
value = value.id
```

Reply:
This won't working because of " elif value " at line 598